File: backend/GscTracking.Api/Models/Expense.cs
namespace GscTracking.Api.Models;

public class Expense
{
    public int Id { get; set; }
    public int JobId { get; set; }
    public Job Job { get; set; } = null!;
    public ExpenseType Type { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public DateTime Date { get; set; } = DateTime.UtcNow;
    public string? ReceiptReference { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/expense.ts
export interface Expense {
  id: number
  jobId: number
  type: ExpenseType
  description: string
  amount: number
  date: string
  receiptReference?: string | null
  createdAt: string
  updatedAt: string
}

export type ExpenseType = 'Parts' | 'Labor' | 'Service'

export interface ExpenseRequestDto {
  type: ExpenseType
  description: string
  amount: number
  date: string
  receiptReference?: string | null
}

export const EXPENSE_TYPES: ExpenseType[] = ['Parts', 'Labor', 'Service']

export const EXPENSE_TYPE_LABELS: Record<ExpenseType, string> = {
  Parts: 'Parts',
  Labor: 'Labor',
  Service: 'Service'
}

export const EXPENSE_TYPE_COLORS: Record<ExpenseType, string> = {
  Parts: 'bg-blue-100 text-blue-800 border-blue-200',
  Labor: 'bg-green-100 text-green-800 border-green-200',
  Service: 'bg-purple-100 text-purple-800 border-purple-200'
}
